### PR TITLE
=str fix Scala 3 signature for Tcp() / AkkaSSLConfig()

### LINKED ISSUE #31266: Scala 3 source incompatibilities around implicit ActorSystem
* `Tcp()` does not work any more but requires an explicit ActorSystem.
 * Automatic conversion between classic and typed ActorSystem does not work implicitly any more, e.g. `Http()` does not work sometimes when only a classic ActorSystem is implicitly in scope.

### PR DESCRIPTION
**Pull request by jrudolph**
This was broken for some reason in #30324 / 3a3e643e074d2e57e320b8efb38bb7fba440e4b3 and not covered by tests.

Fixes #31266

Can later be fixed in https://github.com/akka/akka-http/pull/4079

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala [hunk 1]
```diff
@@ -36,7 +35,7 @@ import akka.util.ByteString
 import akka.util.JavaDurationConverters._
 import akka.util.unused
 
-object Tcp extends ExtensionId[Tcp] with TcpImplicitExtensionIdApply with ExtensionIdProvider {
+object Tcp extends ExtensionId[Tcp] with ExtensionIdProvider {
```
Comment (jrudolph) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Tcp.scala:38]: Technically this is not binary compatible? Should we keep internal class empty around? Or hope that the internal class referenced accidentally?

**Comment by johanandren** [comment]
I think we discussed it during the Scala 3 work and said the implicit extension wasn't a very important feature and that we could skip that for Scala 3 and only have explicit system apply. It still works on Scala 2 right?

**Review by johanandren — APPROVED** [review]
LGTM

**Comment by jrudolph** [comment]
> I think we discussed it during the Scala 3 work and said the implicit extension wasn't a very important feature and that we could skip that for Scala 3 and only have explicit system apply. It still works on Scala 2 right?

Yes, it works, but we want to cross-build and avoid extra updates for us or others.

**Review by patriknw — APPROVED** [review]
LGTM

**Pull request merged**